applications/fishing-map/src/features/map/MapDraw.tsx

```diff
@@ -1,19 +1,297 @@
+          `${name}.json`,
+          {
+            type: 'application/json',
+          }
+        )
+        const dataset = {
+          name,
+          public: true,
+          type: DatasetTypes.Context,
+          category: DatasetCategory.Context,
+          configuration: {
+            format: 'geojson',
+            geometryType: 'draw',
+          } as DatasetConfiguration,
+        }
+        const { payload, error } = await dispatchCreateDataset({
+          dataset,
+          file,
+          createAsPublic: true,
+        })
+        if (error) {
+          console.warn(error)
+        } else if (payload) {
+          addDataviewFromDatasetToWorkspace(payload)
+        }
+        setLoading(false)
+        closeDraw()
+      }
+    },
+    [addDataviewFromDatasetToWorkspace, closeDraw, dispatchCreateDataset]
+  )
+
+  const onSaveClick = useCallback(() => {
+    if (features && features.length > 0 && layerName) {
+      createDataset(features, layerName)
+    }
+  }, [createDataset, features, layerName])
+
+  const editorMode = useMemo(() => {
+    if (drawMode === 'disabled') {
+      return
+    }
+    return drawMode === 'edit' ? new EditingMode() : new DrawPolygonMode()
+  }, [drawMode])
+
+  const hasFeaturesDrawn = features !== undefined && features.length > 0
+  let saveTooltip = ''
+
+  if (!layerName) {
+    saveTooltip = t('layer.nameRequired', 'Layer name is required')
+  } else if (!hasFeaturesDrawn) {
+    saveTooltip = t('layer.geometryRequired', 'Draw a polygon is required')
+  }
+
   return (
-    <Editor
-      clickRadius={12}
-      mode={mode === 'edit' ? EditingMode : DrawPolygonMode}
-      onSelect={(select: any) => {
-        console.log(select)
-      }}
-    />
+    <Fragment>
+      {editorMode && (
+        <Editor
+          ref={editorRef}
+          clickRadius={12}
+          features={features}
+          mode={editorMode}
+          onUpdate={onEditorUpdate}
+          onSelect={onEditorSelect}
+        />
+      )}
+      {editorMode && currentEditHandle && (
+        <Popup
+          latitude={currentEditHandle[1]}
+          longitude={currentEditHandle[0]}
+          closeButton={true}
+          closeOnClick={true}
+          onClose={() => setSelectedEditHandleIndex(null)}
+          className={cx(styles.popup)}
+          captureClick
+        >
+          <div className={styles.popupContent}>
+            <label>{t('common.latitude', 'Latitude')}</label>
+            <InputText
+              value={currentEditHandle[1]}
+              onChange={onHandleLatitudeChange}
+              inputSize="small"
+            />
+            <label>{t('common.longitude', 'longitude')}</label>
+            <InputText
+              value={currentEditHandle[0]}
+              onChange={onHandleLongitudeChange}
+              inputSize="small"
+            />
+          </div>
+        </Popup>
+      )}
+      <div ref={containerRef} className={cx(styles.container, { [styles.hidden]: !editorMode })}>
```

Comment:
I need the dom element to be present and ensure it is mounted so the eventManager recognises it